lib/config/migration.ts

```diff
@@ -110,6 +80,13 @@ export function migrateConfig(config: RenovateConfig): MigratedConfig {
           for (const item of migratedConfig[key] as unknown[]) {
             if (is.object(item) && !is.array(item)) {
               const arrMigrate = migrateConfig(item as RenovateConfig);
+              if (
+                (item as PackageRule).matchDepTypes?.some((depType) =>
+                  depTypes.includes(depType)
+                )
+              ) {
+                delete arrMigrate.migratedConfig.packageRules;
```

Comment:
do we still need this? 🤔

Reply:
It didn't make sense it to me but since it was part of old logic I kept it.

Reply:
Removed the if-block. No effect on tests and example migration PRs [1](https://github.com/RahulGautamSingh-testing/without_old_logic/pull/3) and [2](https://github.com/RahulGautamSingh-testing/wologic2/pull/2/files) 
Cannot find any real usage for it.

Ref PR: https://github.com/renovatebot/renovate/pull/1892